File: php/testcode/benchmark_test_01102.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest01102(BenchmarkRequest $req): BenchmarkResponse {
    $data = json_decode($req->bodyStr(), true);
    $allowed = ['name' => true, 'email' => true, 'phone' => true];
    $safe = array_intersect_key($data, $allowed);
    $pdo = getDbConnection();
    $pdo->prepare("UPDATE users SET name = ?, email = ?, phone = ? WHERE id = ?")
        ->execute([$safe['name'] ?? '', $safe['email'] ?? '', $safe['phone'] ?? '', $req->param('id')]);
    return BenchmarkResponse::ok('User updated');
}

File: rust/testcode/benchmark_test_01102.rs
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let username = req.param("username");

    if !username.chars().all(|c| c.is_alphanumeric()) {
        return super::shared::BenchmarkResponse::bad_request("Invalid username");
    }

    let query = format!("SELECT * FROM users WHERE username = '{}'", username);
    super::shared::BenchmarkResponse::ok(&format!("Executed: {}", query))
}
